libs/execution_context/ebpf_core.c

```diff
@@ -400,9 +405,15 @@ _ebpf_core_protocol_map_find_element(
     if (retval != EBPF_SUCCESS)
         goto Done;
 
-    retval = ebpf_map_find_entry(map, key_length, request->key, value_length, reply->value, 0);
-    if (retval != EBPF_SUCCESS)
-        goto Done;
+    if (request->find_and_delete) {
+        retval = ebpf_map_find_and_delete_entry(map, key_length, request->key, value_length, reply->value, 0);
```

Comment:
The last argument is flags.  Why not just use ebpf_map_find_entry() with a flag set to mean delete if found?